manifests/server.pp

```diff
@@ -1,64 +1,218 @@
+#   When true, the user 'hyperglass' is managed.
+# @param manage_group
+#   When true, the group 'hyperglass' is managed.
+# @param python_version
+#   Specifies which version of python will be used by python::pyvenv if the
+#   `python3_version` fact is not found.
+# @param devices
+#   Hash containing all the devices managed by hyperglass.
+# @param commands
+#   Specific commands that can be used by the devices
+# @param data
+#   Generic hyperglass configuration hash.
 #
 # @see https://hyperglass.io/
 #
-# @author Tim Meusel <tim@bastelfreak.de>
 class hyperglass::server (
   Boolean $manage_depended_services = true,
-  Boolean $manage_python            = true,
-  Boolean $manage_gcc               = true,
-  Hash $data                        = {},
-  Hash $commands                    = {},
-  Hash $devices                     = {
-    'routers' => [
-      {
-        'name' => 'atl_router01',
-        'address' => '10.0.0.2',
-        'network' => {
-          'name' => 'secondary',
-          'display_name' => 'That Other Network',
-        },
-        'credential' => {
-          'username' => 'user2',
-          'password' => ' secret2',
-        },
-        'display_name' => 'Atlanta, GA',
-        'port' => 22,
-        'nos' => 'juniper',
-        'vrfs' => [
-          {
-            'name' => 'default',
-            'display_name' => 'Global',
-            'ipv4' => {
-              'source_address' => '192.0.2.2',
-            },
-          },
-        ],
-      },
-    ],
-  },
+  Boolean $manage_gcc = true,
+  Boolean $manage_node = true,
+  Boolean $manage_nginx = true,
+  Boolean $manage_python = true,
+  Boolean $manage_redis = true,
+  Boolean $manage_yarn = true,
+  Boolean $manage_user = true,
+  Boolean $manage_group = true,
+  String[1] $python_version = '3.6',
+  Hash $data = {},
+  Hash $commands = {},
+  Hash $devices = {},
 ) {
-  unless $facts['os']['name'] in ['CentOS', 'RedHat', 'VirtuozzoLinux'] {
-    fail('the hyperglass::server class currently only works on CentOS/RedHat')
+  if $facts['os']['family'] != 'RedHat' {
+    warning('This module only supports the RedHat family of platforms.')
   }
 
+  include hyperglass::agent
+
   if $manage_depended_services {
-    require hyperglass::server::dependencies
+    if $manage_nginx {
+      include nginx
+    }
+
+    if $manage_python {
+      ensure_resource('class', 'python', { 'version' => '3', 'dev' => 'present' })
+    }
+
+    if $manage_gcc {
+      ensure_resource('package', 'gcc', { 'ensure' => 'installed' })
+    }
+
+    if $manage_redis {
+      class { 'redis::globals':
+        scl    => 'rh-redis5',
+        before => Class['redis'],
+      }
+
+      class { 'redis':
+        manage_repo => true,
+      }
+    }
+
+    if $manage_node {
+      class { 'nodejs':
+        repo_url_suffix => '14.x',
+      }
+    }
+
+    if $manage_yarn {
+      yumrepo { 'yarn':
+        ensure   => 'present',
+        baseurl  => 'https://dl.yarnpkg.com/rpm/',
```

Comment:
how is this supposed to work for non-rpm systems?

Reply:
The module only supports EL7